Amplify/Core/Error/DataError.swift

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Errors associated with data-related operations on Amplify. Commonly
+/// used by the API and DataStore categories.
+enum DataError: Error {
+
+    /// Indicates some data marked as required was unavailable.
+    case dataUnavailable
+
+    /// Indicates that expected data could not be loaded due to the
+    /// lack of integrity.
+    case dataIntegrity(underlyingError: Error? = nil)
+
+    /// Indicates an unexpected scenario happened and the specific
+    /// error could not be resolved.
+    case unknown(ErrorDescription, RecoverySuggestion, Error)
+}
```

Comment:
Is this being mapped to a different error type somewhere within `Core`? I'm sure I'm just missing something, but how is this being used anywhere if it's internal?

Reply:
DataError was introduced by @drochetti in his Custom Selection Set PR (https://github.com/aws-amplify/amplify-swift/pull/2526) which is included in this PR as well. Will remove DataError and keep CoreError to keep it a non-breaking change, more info in https://github.com/aws-amplify/amplify-swift/pull/2583#issuecomment-1372540315

Reply:
removed